tribits/common_tpls/FindTPLNetcdf.cmake

```diff
@@ -44,98 +44,158 @@
+      )
+    if (netCDF_HAS_HDF5)
+      find_package(hdf5 CONFIG REQUIRED)
+      string(APPEND configFileStr
+        "set(hdf5_DIR \"${hdf5_DIR}\")\n" )
+      string(APPEND configFileStr
+        "find_dependency(hdf5 CONFIG)\n"
+        )
+    endif()
+    string(APPEND configFileStr
+      "add_library(Netcdf::all_libs  INTERFACE  IMPORTED  GLOBAL)\n"
+      )
+    string(APPEND configFileStr
+      "target_link_libraries(Netcdf::all_libs  INTERFACE  netCDF::netcdf)\n")
+    set(buildDirExternalPkgsDir
+      "${${PROJECT_NAME}_BINARY_DIR}/${${PROJECT_NAME}_BUILD_DIR_EXTERNAL_PKGS_DIR}")
+    set(tplConfigFile
+      "${buildDirExternalPkgsDir}/Netcdf/NetcdfConfig.cmake")
+    file(WRITE "${tplConfigFile}" "${configFileStr}")
+    # the following gets set by tribits_tpl_find_include_dirs_and_libraries
+    # and it is not well-documented that you need to set it yourself if not using
+    # tribits_tpl_find_include_dirs_and_libraries
+    set(TPL_Netcdf_NOT_FOUND FALSE)
+    message("-- TPL_Netcdf_PARALLEL = netCDF_HAS_PARALLEL = ${netCDF_HAS_PARALLEL}")
+    set(TPL_Netcdf_PARALLEL ${netCDF_HAS_PARALLEL})
+    message("-- TPL_Netcdf_Enables_Netcdf4 = netCDF_HAS_NC4 = ${netCDF_HAS_NC4}")
+    set(TPL_Netcdf_Enables_Netcdf4 ${netCDF_HAS_NC4} CACHE BOOL
       "True if netcdf enables netcdf-4")
-    set(TPL_Netcdf_Enables_PNetcdf ${NetCDF_NEEDS_PNetCDF} CACHE BOOL
-      "True if netcdf enables pnetcdf")
-    set(TPL_Netcdf_PARALLEL ${NetCDF_PARALLEL} CACHE INTERNAL
-      "True if netcdf compiled with parallel enabled")
-    set(TPL_Netcdf_LIBRARY_DIRS ${_hdf5_LIBRARY_SEARCH_DIRS} CACHE PATH
-      "${DOCSTR} library files")
-    set(TPL_Netcdf_LIBRARIES ${NetCDF_LIBRARIES} CACHE PATH
-      "List of semi-colon separated library names (not 'lib' or extension).")
-    set(TPL_Netcdf_INCLUDE_DIRS ${NetCDF_INCLUDE_DIRS} CACHE PATH
-      "${DOCSTR} header files.")
   endif()
+
 else()
-  # Curl library is only required if DAP is enabled; should detect inside
-  # FindNetCDF.cmake, but that is not being called... SEMS has DAP enabled;
-  # many HPC systems don't, but they override these settings...
-  find_program(NC_CONFIG "nc-config")
-  if (NC_CONFIG)
-    execute_process(COMMAND "nc-config --has-dap2"
-                    OUTPUT_VARIABLE NETCDF_HAS_DAP2)
-    execute_process(COMMAND "nc-config --has-dap4"
-                    OUTPUT_VARIABLE NETCDF_HAS_DAP4)
+
+  set(REQUIRED_HEADERS netcdf.h)
+  set(REQUIRED_LIBS_NAMES netcdf)
+
+  if (TPL_ENABLE_MPI)
+    set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} pnetcdf)
   endif()
-  if ((NOT NC_CONFIG) OR NETCDF_HAS_DAP2 OR NETCDF_HAS_DAP4)
-    set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} curl)
+
+  #
+  # Second, search for Netcdf components (if allowed) using the standard
+  # find_package(NetCDF ...).
+  #
+  tribits_tpl_allow_pre_find_package(Netcdf  Netcdf_ALLOW_PREFIND)
+  if (Netcdf_ALLOW_PREFIND)
+
+    message("-- Using find_package(Netcdf ...) ...")
+
+    set(CMAKE_MODULE_PATH
```

Comment:
You could use `list(APPEND)` here.

Reply:
Because this file is so hard to fully test in all of the use cases, I am not going to change this code at this point.  (In the future, it should all be stripped out.)